### ISSUE TITLE
Application is broken after upgrading to 0.10.2

### ISSUE THREAD

**Issue by shinayser** [flutter/flutter#24055]
I've just updated to 0.10.2 (my previous version was 0.9.4)
I have a button that opens a native Activity when taped. Then, after closing this native code activity (by pressing the back button), the applications crashes on simulator (android pie).

```
W/ActivityThread( 9965): handleWindowVisibility: no activity for token android.os.BinderProxy@6a6e0c2
W/.denox.flow.de( 9965): Accessing hidden method Landroid/view/View;->computeFitSystemWindows(Landroid/graphics/Rect;Landroid/graphics/Rect;)Z (light greylist, reflection)
W/.denox.flow.de( 9965): Accessing hidden method Landroid/view/ViewGroup;->makeOptionalFitsSystemWindows()V (light greylist, reflection)
I/Choreographer( 9965): Skipped 67 frames!  The application may be doing too much work on its main thread.
I/OpenGLRenderer( 9965): Davey! duration=1131ms; Flags=0, IntendedVsync=2017433142060, Vsync=2018549808682, OldestInputEvent=9223372036854775807, NewestInputEvent=0, HandleInputStart=2018555761760, AnimationStart=2018555811110, PerformTraversalsStart=2018560401110, DrawStart=2018560677160, SyncQueued=2018561463670, SyncStart=2018561744860, IssueDrawCommandsStart=2018562111380, SwapBuffers=2018562361720, FrameCompleted=2018564966960, DequeueBufferDuration=325000, QueueBufferDuration=472000, 
D/EGL_emulation( 9965): eglMakeCurrent: 0x74d14a8fd2e0: ver 3 0 (tinfo 0x74d13b31d500)
W/ActivityThread( 9965): handleWindowVisibility: no activity for token android.os.BinderProxy@f7c59f6
D/EGL_emulation( 9965): eglMakeCurrent: 0x74d14a8fd2e0: ver 3 0 (tinfo 0x74d13b31d500)
D/EGL_emulation( 9965): eglCreateContext: 0x74d148fb7ba0: maj 3 min 0 rcv 3
D/EGL_emulation( 9965): eglMakeCurrent: 0x74d148fb7ba0: ver 3 0 (tinfo 0x74d13b20ec40)
D/EGL_emulation( 9965): eglMakeCurrent: 0x74d148fb7ba0: ver 3 0 (tinfo 0x74d14a92dd80)
D/skia    ( 9965): Program linking failed.
E/emuglGLESv2_enc( 9965): device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glLinkProgram:1525 GL error 0x501
E/emuglGLESv2_enc( 9965): device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glGetProgramBinary:4058 GL error 0x502
D/EGL_emulation( 9965): eglMakeCurrent: 0x74d14a8fd2e0: ver 3 0 (tinfo 0x74d13b31d500)
Lost connection to device.
```

On an Android device (android nougat) the app does not crash, but gets visually broken. If I rotate the device,it goes back to normal.

![flutter_01](https://user-images.githubusercontent.com/1146982/48136087-e8427f80-e285-11e8-837c-06a19a134e4c.png)
![flutter_02](https://user-images.githubusercontent.com/1146982/48136091-ebd60680-e285-11e8-9d29-150c6016ef06.png)

**Comment by zoechi** [comment]
Please add error, log and `flutter doctor` output as text, not as screenshot.
Screenshots are not searchable.

**Comment by eseidelGoogle** [comment]
The visual glitches you see on Android devices look like: https://github.com/flutter/flutter/issues/24216.  Do you agree @liyuqian?

The crash you list:
```
E/emuglGLESv2_enc( 9965): device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glLinkProgram:1525 GL error 0x501
E/emuglGLESv2_enc( 9965): device/generic/goldfish-opengl/system/GLESv2_enc/GL2Encoder.cpp:s_glGetProgramBinary:4058 GL error 0x502
D/EGL_emulation( 9965): eglMakeCurrent: 0x74d14a8fd2e0: ver 3 0 (tinfo 0x74d13b31d500)
```
sounds new to me but is less concerning if its' emulator-only.

**Comment by eseidelGoogle** [comment]
https://github.com/flutter/flutter/issues/22568 has similar emulator logs.

**Comment by eseidelGoogle** [comment]
My guess (hope?) is that both of the issues you've seen are fixed on master, but we should confirm.

**Comment by shinayser** [comment]
I can confirm that this issue has been fixed on the current master (0.11.9).

**Comment by liyuqian** [comment]
Thank you for figuring this out before I have time to take a look! Yes, it looks like #24216 and it's nice to hear that it's fixed along with #24216 .
